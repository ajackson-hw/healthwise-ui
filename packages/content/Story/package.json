{
  "name": "@healthwise/content-story",
  "version": "2.0.0",
  "author": "Healthwise",
  "description": "Healthwise UI Assets - Story structured content component",
  "license": "MIT",
  "homepage": "https://github.com/healthwise/healthwise-ui#readme",
  "main": "index.js",
  "publishConfig": {
    "access": "public"
  },
  "dependencies": {
    "classnames": "^2.2.5",
    "prop-types": "^15.5.7"
  },
  "peerDependencies": {
    "@healthwise/theme": "^4.0.0",
    "react": ">=15.4.2"
  }
}
