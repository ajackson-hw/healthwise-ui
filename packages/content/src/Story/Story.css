.story :global(.HwImageWrapper) {
  float: right;
  margin: 0 0 0 20px;
  background: var(--color-background-contrast-light);
  padding: 10px 10px 30px;
  line-height: 1;
}
.story :global(.HwImageWrapper) img {
  border: 3px solid var(--color-border);
}

.story :global(.HwNote) {
  float: right;
  clear: both;
  margin-bottom: 20px;
  font-style: italic;
  padding-right: 10px;
  margin-top: -30px;
  color: var(--color-text-primary);
}

.story :global(.HwNote) ~ :global(.HwNote) {
  display: none;
}
