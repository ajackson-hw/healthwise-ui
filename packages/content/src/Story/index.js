import React from 'react'
import PropTypes from 'prop-types'
import { withTheme, createGlobalStyle } from 'styled-components'

import { defaultTheme } from '@healthwise-ui/core/Theme'

const GlobalStyle = createGlobalStyle`
  .HwImageWrapper {
    float: right;
    margin: 0 0 0 20px;
    background: ${props => props.theme.colorBackgroundContrastLight};
    padding: 10px 10px 30px;
    line-height: 1;
  }

  .HwImageWrapper img {
    border: 3px solid ${props => props.theme.colorBorder};
  }

  .HwNote {
    float: right;
    clear: both;
    margin-bottom: 20px;
    font-style: italic;
    padding-right: 10px;
    margin-top: -30px;
    color: ${props => props.theme.colorTextPrimary};
  }

  .HwNote ~ .HwNote {
    display: none;
  }

`

class Story extends React.Component {
  render() {
    let { data, ...otherProps } = this.props

    return (
      <div
        className="hw-content-story"
        dangerouslySetInnerHTML={{ __html: data.html }}
        {...otherProps}
      >
        <GlobalStyle theme={this.props.theme} />
      </div>
    )
  }
}

Story.propTypes = {
  data: PropTypes.shape({
    html: PropTypes.string,
  }),
}

Story.defaultProps = {
  theme: defaultTheme,
}

export default withTheme(Story)
