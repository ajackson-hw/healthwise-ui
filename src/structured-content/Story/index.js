import React from 'react'
import PropTypes from 'prop-types'

import styles from './Story.css'

class Story extends React.Component {
  render() {
    let { data, ...otherProps } = this.props
    const outerClasses = 'hw-content-story ' + styles.story

    return (
      <div
        className={outerClasses}
        dangerouslySetInnerHTML={{ __html: data.html }}
        {...otherProps}
      />
    )
  }
}

Story.propTypes = {
  data: PropTypes.shape({
    html: PropTypes.string,
  }),
}

export default Story
