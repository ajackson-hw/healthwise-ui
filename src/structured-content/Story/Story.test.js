/* global describe, it, expect, jest */
import React from 'react'
import renderer from 'react-test-renderer'

import Story from './index'

describe('<Story />', () => {
  it('renders correctly with defaults', () => {
    const component = renderer.create(<Story data={{ html: '<b>bold</b>' }} />)
    expect(component).toMatchSnapshot()
  })
})
