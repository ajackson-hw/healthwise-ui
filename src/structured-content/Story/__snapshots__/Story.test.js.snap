// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Story /> renders correctly with defaults 1`] = `
<div
  className="hw-content-story story"
  dangerouslySetInnerHTML={
    Object {
      "__html": "<b>bold</b>",
    }
  }
/>
`;
